python/pyspark/sql/functions.py

```diff
@@ -2183,14 +2183,29 @@ def pandas_udf(f=None, returnType=StringType()):
     :param f: python function if used as a standalone function
     :param returnType: a :class:`pyspark.sql.types.DataType` object
 
-    # TODO: doctest
+    >>> from pyspark.sql.types import IntegerType, StringType
+    >>> slen = pandas_udf(lambda s: s.str.len(), IntegerType())
+    >>> @pandas_udf(returnType=StringType())
```

Comment:
Have we installed pyarrow on Jenkins? The failed test complains `ImportError: No module named pyarrow`.

Reply:
We could just do `  # doctest: +SKIP` maybe.

Reply:
Hmm, I thought that the Jenkins environment for unit tests would be the same for doctests and have pyarrow installed. @holdenk or @shaneknapp do you know if that is the case?

Reply:
adding @JoshRosen too.

the doc building node (amp-jenkins-worker-01) doesn't have arrow installed for the default conda python 2.7 environment.  for the python 3 environment, we're running arrow 0.4.0.

i looked at the script and it seems to be agnostic to python 2 vs 3...   once i know which version of python we'll be running i can make sure that the version of arrow installed is correct.

Reply:
Cool, thanks @shaneknapp!

Reply:
Hm.. but shouldn't we skip those doctests because they are not hard dependencies anyway?

Reply:
That's true, I see that `toPandas()` also skips doctests. I'll skip this now and can always enable later if we decide differently.  @shaneknapp , looks like we will hold off on this so no need to do anything to Jenkins I believe, sorry to bug you.